src/test/groovy/edu/ie3/datamodel/utils/GridAndGeoUtilsTest.groovy

```diff
@@ -123,4 +128,42 @@ class GridAndGeoUtilsTest extends Specification {
 		expect:
 		GridAndGeoUtils.getCoordinateDistances(basePoint, points) == new TreeSet(coordinateDistances)
 	}
+
+	def "TotalLengthOfLineString correctly calculates the total length of lineString correctly"() {
+		given:
+		LineString lineString = GeoUtils.DEFAULT_GEOMETRY_FACTORY.createLineString([
+			new Coordinate(22.69962d, 11.13038d, 0),
+			new Coordinate(20.84247d, 28.14743d, 0),
+			new Coordinate(24.21942d, 12.04265d, 0)] as Coordinate[])
+
+		when:
+		ComparableQuantity<Length> y = GridAndGeoUtils.totalLengthOfLineString(lineString)
+
+		then:
+		y.isGreaterThanOrEqualTo(Quantities.getQuantity(3463.37 - 10, KILOMETRE))
```

Comment:
You may use `QuantityUtil` with its tolerance methods.

Reply:
done :)